### ISSUE TITLE
`get.opp.files` doesn't properly remove outliers

### ISSUE THREAD

**Issue by ctberthiaume** [armbrustlab/popcycle#52]
Using HOT-294 as an example:
```R
# Get the outlier table
outlier_table <- get.outlier.table(db)
# Get all unique file IDs in the the opp table
all_opp_files <- unique(get.opp.table(db)$file)
# Get OPP files with `get.opp.files`, which should remove outliers by default
opp_files <- get.opp.files(db)
# Confirm that the first file in the outlier table is an outlier
outlier_table[1,]
#                                file flag
#1 2017_170/2017-06-19T22-38-43+00-00    1
# This file should not be in `opp_files`, but it is
opp_files[1]
#[1] "2017_170/2017-06-19T22-38-43+00-00"
```

I think this is because the outlier matching in `get.opp.files` isn't using the match function correctly.

**Comment by ctberthiaume** [comment]
This can be pused down to SQL, and probably should be. One major advantage of a declarative language like SQL is these kinds of matching bugs don't happen. Rough cut at a SQL query to only get OPP files that are in the SFL table and have data in all three quantiles. Could add another join to outliers table too. 
```sql
SELECT sfl.date AS date,
       opp.*
FROM sfl
INNER JOIN opp
ON sfl.file == opp.file
INNER JOIN
(
  SELECT file FROM opp
  WHERE opp_count > 0
  GROUP BY file
  HAVING count(file) == 3
) AS o
ON sfl.file == o.file
ORDER BY sfl.date ASC;

```

**Comment by ctberthiaume** [comment]
Fixed in 1e8ef17.
